owtf/api/handlers/auth.py

```diff
@@ -0,0 +1,230 @@
+"""
+owtf.api.handlers.auth
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+"""
+from owtf.api.handlers.base import APIRequestHandler
+from owtf.lib.exceptions import APIError
+from owtf.models.user import User
+from datetime import datetime, timedelta
+import tornado
+import bcrypt
+import json
+import base64
+import uuid
+import jwt
+import re
+
+JWT_SECRET = 'my_secret_key'
+JWT_ALGORITHM = 'HS256'
+JWT_EXP_DELTA_SECONDS = 60 * 60 * 24
+
+
+class LogInHandler(APIRequestHandler):
+    """LogIn using the correct credentials (email, password). After successfull login a JWT Token is generated"""
+
+    SUPPORTED_METHODS = ["POST"]
+
+    def post(self):
+        """
+            **Example request**:
+
+            .. sourcecode:: http
+
+            POST /api/v1/login/ HTTP/1.1
+
+            **Example response**:
+
+            .. sourcecode:: http
+
+            **Login successful response**;
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "ok",
+                    "jwt-token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxOCwiZXhwIjoxNjE1NjU0OTk1fQ.wSMr5laDNVTxuTTJI4FWssnZa2xYL4PS8llDLDB4eyY"
+                }
+            }
+
+            **Login failed response**;
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "fail",
+                    "message": "Invalid login credentials"
+                }
+            }
+
+        """
+        body_data = json.loads(self.request.body.decode('utf-8'))
+        email = body_data['email']
```

Comment:
in order for this to work correctly, you should catch exceptions gracefully,

```
email = body_data.get('email', None)
if email is not None:
```

Reply:
![image](https://user-images.githubusercontent.com/37950602/121947338-91718080-cd73-11eb-99fa-07d4f4ad4922.png)
Handled it at all the required locations.